src/compiler/checker.ts

```diff
@@ -12057,6 +12057,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (!links.type) {
             Debug.assertIsDefined(links.deferralParent);
             Debug.assertIsDefined(links.deferralConstituents);
+            if (links.deferralParent === neverType) {
+                const functionDecl = links.jsSyntheticRestParameterFunctionDeclaration!;
+                if (isFunctionExpressionOrArrowFunction(functionDecl)) {
+                    const contextualSignature = getContextualSignature(functionDecl);
+                    if (contextualSignature) {
+                        return getRestTypeAtPosition(contextualSignature, functionDecl.parameters.length);

```

Comment:
Is it intentional that this isn’t assigned to `links.type`?

Reply:
It's intentional in the sense that I followed what some other, similar, situation was already doing. At the time of writing this, I knew that I'm not assigning to `links.type` and I've decided not to do it... but now I'm not 100% sure why I didn't do it.

Taking a *quick* look at related functions I might have been trying to replicate he intention of the comment in `getTypeOfVariableOrParameterOrProperty`. OTOH, that still assigns `links.type` - just conditionally. But yeee, I think it was to make sure that the type set by `assignParameterType` prevails.

Reply:
`getTypeOfVariableOrParameterOrProperty` only assigns conditionally because there is a `checkMode`, and we only cache types computed with `CheckMode.Normal`. This function appears to be a pure function of `symbol` and so AFAIK its type can/should be cached to prevent repeating work next time it’s asked for.